packages/react/src/auto/shadcn/ShadcnAutoForm.tsx

```diff
@@ -0,0 +1,122 @@
+import type { ActionFunction } from "@gadgetinc/api-client-core";
+import React, { ComponentProps } from "react";
+import { FormProvider } from "../../useActionForm.js";
+import { humanizeCamelCase, type OptionsType } from "../../utils.js";
+import type { AutoFormProps } from "../AutoForm.js";
+import { useAutoForm } from "../AutoForm.js";
+import { validateAutoFormProps } from "../AutoFormActionValidators.js";
+import { AutoFormMetadataContext } from "../AutoFormContext.js";
+import type { FormProps, ShadcnElements } from "./elements.js";
+import { makeShadcnAutoInput } from "./inputs/ShadcnAutoInput.js";
+import { makeShadcnAutoSubmit } from "./submit/ShadcnAutoSubmit.js";
+import { makeSubmitResultBanner } from "./submit/ShadcnSubmitResultBanner.js";
+
+/**
+ * Renders a form for an action on a model automatically using Shadcn
+ */
+export const makeAutoForm =
+  <Elements extends ShadcnElements>({ Form, Input, Button, Alert, Skeleton, AlertTitle, AlertDescription, Label, Checkbox }: Elements) =>
+  <GivenOptions extends OptionsType, SchemaT, ActionFunc extends ActionFunction<GivenOptions, any, any, SchemaT, any>>(
+    props: AutoFormProps<GivenOptions, SchemaT, ActionFunc> & ComponentProps<typeof Form>
+  ) => {
```

Comment:
just realizing that we should probably use named functions instead of anonymous functions in these factories here so that stack traces and the React dev tools can use the right name for the components in development